packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -78,26 +81,56 @@ export const styles = css`
     }
 
     .control:hover {
-        box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-        outline: none;
+        border-color: ${borderHoverColor};
+        box-shadow: 0px 0px 0px 1px ${borderHoverColor} inset;
+        transition: box-shadow ${mediumDelay};
     }
 
     .control${focusVisible} {
+        border-color: ${borderHoverColor};
         box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-        outline: ${borderWidth} solid ${borderHoverColor};
-        outline-offset: -4px;
     }
 
     .control:active {
         box-shadow: none;
         outline: none;
+        transition: outline ${mediumDelay};
+        transition: box-shadow ${mediumDelay};
+    }
+
+    .control::before {
+        content: '';
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        pointer-events: none;
+        box-sizing: border-box;
+        outline: none;
+        background-clip: content-box;
+        z-index: -1;
```

Comment:
what's the z-index: -1 about?

Reply:
Since the `::before` class has backgrounds in some conditions (like `:active`), if the z-index didn't exist, the background would overlap on top of the button text and cause the text color to look off.